### PR TITLE
Make PGLSFilter interface more flexible (for cephfs)

### PR DESCRIPTION
**Pull request by jcsp**

### CODE REVIEW

**Comment by jcsp** [comment]
@athanatos ping -- who's a good person to review this?

**Comment by liewegas** [comment]
On Wed, 5 Aug 2015, John Spray wrote:
> @athanatos ping -- who's a good person to review this?

I've just been waiting until the wip-newstore-sort is sorted out, but 
that's basically ready.  I'll take a look at this today!

**Comment by liewegas** [comment]
Oh nevermind, thought were were talking about the other PR.  I'll look at this one too though :)

File: src/osd/ReplicatedPG.cc [hunk 1]
```diff
@@ -504,7 +504,8 @@ void ReplicatedPG::wait_for_blocked_object(const hobject_t& soid, OpRequestRef o
   op->mark_delayed("waiting for blocked object");
 }
 
-bool PGLSParentFilter::filter(bufferlist& xattr_data, bufferlist& outdata)
+bool PGLSParentFilter::filter(const std::string &obj_name,
```
Comment (liewegas) [inline src/osd/ReplicatedPG.cc]: I'd pass in the hobject_t instead of just the name portion

File: src/osd/ReplicatedPG.h [hunk 2]
```diff
@@ -63,8 +63,20 @@ class PGLSFilter {
...
+   * xattr key, or empty string.  If non-empty, this xattr will be fetched
+   * and the value passed into ::filter
+   */
+   virtual string& get_xattr() { return xattr; }
+
+  /**
+   * If true, objects without the named xattr (if xattr name is not empty)
+   * will be rejected without calling ::filter
+   */
+  virtual bool reject_empty() { return true; }
```
Comment (liewegas) [inline src/osd/ReplicatedPG.h]: reject_empty_xattr() ?

**Comment by liewegas** [comment]
Two minor nits, otherwise looks good!

**Comment by jcsp** [comment]
Updated this, will merge when gitbuilders are happy

**Pull request merged**